ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-TobaccoUse.xsl

```diff
@@ -0,0 +1,182 @@
+    
+    <xd:doc scope="stylesheet">
+        <xd:desc>Converts ADA tabak_gebruik to FHIR resource conforming to profile nl-core-TobaccoUse</xd:desc>
+    </xd:doc>
+    
+    <xd:doc>
+        <xd:desc>Create a nl-core-TobaccoUse instance as a Observation FHIR instance from ADA tabak_gebruik.</xd:desc>
+        <xd:param name="in">ADA element as input. Defaults to self.</xd:param>
+    </xd:doc>
+    <xsl:template match="tabak_gebruik" name="nl-core-TobaccoUse" mode="nl-core-TobaccoUse" as="element(f:Observation)">
+        <xsl:param name="in" as="element()?" select="."/>
+        <xsl:param name="subject" select="patient/*" as="element()?"/>
+        
+        <xsl:for-each select="$in">
+            <Observation>
+                <xsl:call-template name="insertLogicalId"/>
+                <meta>
+                    <profile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-TobaccoUse"/>
+                </meta>
+                <status value="final"/>
+                <code>
+                    <coding>
+                        <system value="http://snomed.info/sct"/>
+                        <code value="365980008"/>
+                        <display value="bevinding betreffende tabakgebruik en blootstelling aan tabaksrook"/>
+                    </coding>
+                </code>
+                
+                <xsl:for-each select="$subject">
+                    <xsl:call-template name="makeReference">
+                        <xsl:with-param name="in" select="$subject"/>
+                        <xsl:with-param name="wrapIn" select="'subject'"/>
+                    </xsl:call-template>
+                </xsl:for-each>
+                
+                <xsl:if test="waarneming_gebruik/start_datum or waarneming_gebruik/stop_datum">
+                    <effectivePeriod>
+                        <xsl:if test="waarneming_gebruik/start_datum">
+                            <start>
+                                <xsl:attribute name="value">
+                                    <xsl:call-template name="format2FHIRDate">
+                                        <xsl:with-param name="dateTime" select="xs:string(waarneming_gebruik/start_datum/@value)"/>
+                                    </xsl:call-template>
+                                </xsl:attribute>
+                            </start>
+                        </xsl:if>
+                        <xsl:if test="waarneming_gebruik/stop_datum">
+                            <end>
+                                <xsl:attribute name="value">
+                                    <xsl:call-template name="format2FHIRDate">
+                                        <xsl:with-param name="dateTime" select="xs:string(waarneming_gebruik/stop_datum/@value)"/>
+                                    </xsl:call-template>
+                                </xsl:attribute>
+                            </end>
+                        </xsl:if>
+                    </effectivePeriod>
```

Comment:
Ik zou er dit van maken:
```
<effectivePeriod>
                        <xsl:for-each select="waarneming_gebruik/start_datum">
                            <start>
                                <xsl:attribute name="value">
                                    <xsl:call-template name="format2FHIRDate">
                                        <xsl:with-param name="dateTime" select="@value"/>
                                    </xsl:call-template>
                                </xsl:attribute>
                            </start>
                        </xsl:for-each>
                        <xsl:for-each select="waarneming_gebruik/stop_datum">
                            <end>
                                <xsl:attribute name="value">
                                    <xsl:call-template name="format2FHIRDate">
                                        <xsl:with-param name="dateTime" select="@value"/>
                                    </xsl:call-template>
                                </xsl:attribute>
                            </end>
                        </xsl:for-each>
                    </effectivePeriod>
```
Maar is effectief hetzelfde.

Reply:
Aangepast. Is wel consistenter.